### PR TITLE
ISPN-12899 add protostream to spring starter docs

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-12899

### CODE REVIEW

**Review by karesti — APPROVED** [review]

File: documentation/src/main/asciidoc/topics/proc_spring_boot_serialization.adoc [hunk 1]
```diff
@@ -1,16 +1,34 @@
+[id='spring-boot-serialization_{context}']
 = Configuring Marshalling
-Configure {brandname} servers to use Java serialization to marshall objects.
+Configure {brandname} to marshall Java objects into binary format so they can be transferred over the wire or stored to disk.
 
-By default {brandname} server uses a ProtoStream serialization library as the
-default marshaller. However, the ProtoStream marshaller is not supported for
-Spring integration. For this reason you should use the Java Serialization Marshaller.
+By default {brandname} uses a Java Serialization marshaller, which requires you to add your classes to an allow list.
+As an alternative you can use ProtoStream.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/topics/proc_spring_boot_serialization.adoc:6]: ProtoStream also requires annotating key/value classes with protostream annotations and adding the protostream-processor dependency. Maybe we should link to the protostream marshalling chapter in the dev guide.

**Review by danberindei — COMMENTED** [review]
I have a few post-merge comments @oraNod

File: documentation/src/main/asciidoc/topics/proc_adding_embeddedcm_bean.adoc [hunk 2]
```diff
@@ -1,18 +1,20 @@
-= Adding the EmbeddedCacheManager Bean 
+[id='adding-embeddedcm-bean_{context}']
+= Adding the EmbeddedCacheManager Bean
+The {brandname} starter operates in Remote Client/Server mode with `infinispan-spring-boot-starter-remote` on the classpath by default.
+To use embedded caches, you need to add the `EmbeddedCacheManager` bean.
```
[the remark this answers is not in the document]
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_adding_embeddedcm_bean.adoc:4]: @danberindei I've reworded this and put the changes in a new PR. It's probably less confusing to users to just drop that statement about operating in Remote Client/Server mode.

File: documentation/src/main/asciidoc/topics/proc_spring_boot_serialization.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_spring_boot_serialization.adoc:6]: Added to the new PR: https://github.com/infinispan/infinispan/pull/9217/files#diff-1fea95fa62e0399a3e9694ddc76b8da26a5d7a1861f6d10b89591350cdc1ed84R6

File: documentation/src/main/asciidoc/topics/proc_configuring_remotecm_bean.adoc [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
 +
 [source,java,options="nowrap"]
 ----
 include::code_examples/AutowiredRemoteCacheManager.java[]
 ----
 
+== Properties files
+
+You can specify properties in either `hotrod-client.properties` or `application.properties`.
+The starter first tries to find the server location in `hotrod-client.properties` and then from `application.properties`.
```
[the remark this answers is not in the document]
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_configuring_remotecm_bean.adoc:15]: Added to the new PR here: https://github.com/infinispan/infinispan/pull/9217/files#diff-2c0416b7137e400e1527d2a137ed890c1ea9bd1ec0dd95315f604c0cd533271eR18 

Please add any follow up comments over there. Cheers.

**Pull request merged**